topostats/run_topostats.py

```diff
@@ -303,186 +172,170 @@ def process_scan(
+            plot_and_save(grains.images["gaussian_filtered"], **plotting_config["plot_dict"][plot_name])
 
             plot_name = "z_threshed"
-            PLOT_DICT[plot_name]["output_dir"] = Path(_output_dir)
+            plotting_config["plot_dict"][plot_name]["output_dir"] = Path(_output_dir)
             plot_and_save(
                 grains.images["gaussian_filtered"],
                 filename=filtered_image.filename + "_processed",
-                **PLOT_DICT[plot_name],
+                **plotting_config["plot_dict"][plot_name],
             )
 
             for direction, image_arrays in grains.directions.items():
                 output_dir = Path(_output_dir) / filtered_image.filename / "grains" / f"{direction}"
                 for plot_name, array in image_arrays.items():
-                    PLOT_DICT[plot_name]["output_dir"] = output_dir
-                    plot_and_save(array, **PLOT_DICT[plot_name])
+                    plotting_config["plot_dict"][plot_name]["output_dir"] = output_dir
+                    plot_and_save(array, **plotting_config["plot_dict"][plot_name])
                 # Make a plot of coloured regions with bounding boxes
-                PLOT_DICT["bounding_boxes"]["output_dir"] = output_dir
+                plotting_config["plot_dict"]["bounding_boxes"]["output_dir"] = output_dir
                 plot_and_save(
                     grains.directions[direction]["coloured_regions"],
-                    **PLOT_DICT["bounding_boxes"],
+                    **plotting_config["plot_dict"]["bounding_boxes"],
                     region_properties=grains.region_properties[direction],
                 )
-                PLOT_DICT["coloured_boxes"]["output_dir"] = output_dir
+                plotting_config["plot_dict"]["coloured_boxes"]["output_dir"] = output_dir
                 plot_and_save(
                     grains.directions[direction]["labelled_regions_02"],
-                    **PLOT_DICT["coloured_boxes"],
+                    **plotting_config["plot_dict"]["coloured_boxes"],
                     region_properties=grains.region_properties[direction],
                 )
 
-            plot_name = "mask_overlay"
-            PLOT_DICT[plot_name]["output_dir"] = Path(_output_dir)
-            plot_and_save(
-                grains.images["gaussian_filtered"],
-                filename=filtered_image.filename + "_processed_masked",
-                data2=grains.directions[mask_direction]["removed_small_objects"],
-                **PLOT_DICT[plot_name],
-            )
-
+                plot_name = "mask_overlay"
+                plotting_config["plot_dict"][plot_name]["output_dir"] = Path(_output_dir)
+                plot_and_save(
+                    grains.images["gaussian_filtered"],
+                    filename=filtered_image.filename + "_processed_masked",
+                    data2=grains.directions[direction]["removed_small_objects"],
```

Comment:
Previously this was out of the loop so that only the mask_overlay image would have the effect of the config-specified direction which it no longer does. 

However, we can move this out the loop and call the direction specified in the grainstats_config:
```suggestion
                data2=grains.directions[grainstats_config["direction"]]["removed_small_objects"]
```